### ISSUE TITLE
NNAPI support for torch.cat([...]) when dim=-1

### ISSUE THREAD

**Issue by rubengrab** [pytorch/pytorch#56007]
## 🚀 Feature
NNAPI model should support concatenation when dim=-1

## Motivation

In both **PyTorch** & **TorchScript** format, using torch.cat([...], dim=-1) runs succesfully. But although the conversion from **TorchScript** to **NNAPI** is succesfull, the model fails to run on the **Android device** when axis is **NOT** >=0 for the concatenation opperation.

## Pitch

Using torch.cat([...], dim=-1) in python code (as in: passing _-1_ to _dim_) should also be supported when running model on the Android device.

## Alternatives

An alternative behaviour would be the conversion from TorchScript to NNAPI to fail with a message saying that **only dim >=0 is currently supported**.
